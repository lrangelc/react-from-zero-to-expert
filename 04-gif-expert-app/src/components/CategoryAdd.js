import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const CategoryAdd = (props) => {
  const [inputValue, setInputValue] = useState('Hola Mundo');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 0) {
      // props.setCategories([...props.categories, `${inputValue}`]);
      props.setCategories((categories) => [...categories, `${inputValue}`]);
      setInputValue('');
    }
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <input type='text' value={inputValue} onChange={handleInputChange}></input>
      </form>
    </>
  );
};

CategoryAdd.propTypes = {
  setCategories: PropTypes.func.isRequired
};