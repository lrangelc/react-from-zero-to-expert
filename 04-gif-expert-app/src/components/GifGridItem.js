import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ img, title, url }) => {
  return (
    <div className='card animate__animated animate__fadeIn animate__delay-2s'>
      <img src={url ?? img.url} alt={title ?? img.title}></img>
      <p>{title ?? img.title}</p>
    </div>
  );
};

GifGridItem.propTypes = {
  // title: PropTypes.string.isRequired,
  // url: PropTypes.string.isRequired,
  img: PropTypes.object.isRequired,
};

GifGridItem.defaultProps = {
  img: { url: '', title: '' },
};
