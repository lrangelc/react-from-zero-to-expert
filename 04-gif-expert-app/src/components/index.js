export * from './CategoryAdd';
export * from './GifGrid';
export * from './GifGridItem';