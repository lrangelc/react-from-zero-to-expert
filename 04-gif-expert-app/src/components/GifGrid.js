import React, { useState, useEffect } from 'react';

import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category, index }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url = `https://api.giphy.com/v1/gifs/search?limit=10&q=${category}&api_key=dUaEpCHLvbS1VDigGQT0Zp6ge5yepPT5`;

    const resp = await fetch(url);
    const { data } = await resp.json();
    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url,
      };
    });
    console.log(gifs);
    setImages(gifs);
  };

  return (
    <>
      <h3 key={`h3${index}`}>{category}</h3>
      <div className='card-grid'>
        {images.map((img) => (
          <GifGridItem key={img.id} img={img}></GifGridItem>
        ))}
      </div>
    </>
  );
};
