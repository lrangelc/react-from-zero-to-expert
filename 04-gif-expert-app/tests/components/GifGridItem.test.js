import { render } from '@testing-library/react';
import React from 'react';

import { GifGridItem } from '../../src/components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {
  const title = 'Un titulo';
  const url = 'https://localhost/algo.jpg';
  const img = {
    title,
    url,
  };

  test('debe de mostrar el componente correctamente', () => {
    const { container } = render(<GifGridItem img={img} />);
    expect(container).toMatchSnapshot();
  });
});
