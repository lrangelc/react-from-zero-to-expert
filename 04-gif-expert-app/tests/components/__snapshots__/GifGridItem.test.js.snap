// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div>
  <div
    class="card animate__animated animate__fadeIn animate__delay-2s"
  >
    <img
      alt="Un titulo"
      src="https://localhost/algo.jpg"
    />
    <p>
      Un titulo
    </p>
  </div>
</div>
`;
